### PR TITLE
added spinner

### PR DESCRIPTION
**Pull request by marcosmol204**
Added spinner as required in #83

### CODE REVIEW

**Comment by goldbergyoni** [comment]
image

@mikicho Why like this?

File: src/code-generator/entry-points/non-interactive-cli.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { AppError } from "../error-handling";
 import { factorDefaultOptions } from "../generation-logic/generation-options";
 import { generateApp } from "../generation-logic/generate-service";
+import { spinner, cliTexts } from "../utils";
```
Comment (goldbergyoni) [inline src/code-generator/entry-points/non-interactive-cli.ts:4]: Two notes here:

1. Our CLI has two main 'layers': entry-point (UI) and logic. These 'utils' are not generic rather ui-elements, hence they should get placed within the entry-points folder
2. I would avoid names like 'utils'. Think about the reader of the code, 'utils' kinda tell her - there are multiple things here which you'll have to find out yourself. Consider a name that actually tells what's in there like 'ui-elements'. You may also break the entry-point to two sub-folders: interactiveCLI and nonInteractiveCLI (what we handle now), then put all of those files in the nonInteractiveCLI

**Review by goldbergyoni — CHANGES_REQUESTED** [review]
@marcosmol204 Love this PR 💜

**Comment by mikicho** [comment]
> image
> 
> @mikicho Why like this?

IDK. maybe something has been changed on snyk side? @lirantal (from some reason I can't tag him here)

image

**Comment by lirantal** [comment]
@mikicho this is using the free tier and that you have put the SNYK API TOKEN in this repo's secrets settings, right? can you go to the usage area and check if you've hit the limit on the free plan?

File: package.json [hunk 2]
```diff
@@ -59,13 +59,15 @@
...
     "ink-text-input": "^4.0.3",
+    "ora": "5.4.1",
     "prompts": "^2.4.2",
     "react": "^17.0.2"
   },
   "devDependencies": {
     "@types/axios": "^0.14.0",
     "@types/figlet": "^1.5.4",
     "@types/jest": "^27.4.1",
+    "@types/ora": "^3.2.0",
```
[the remark this answers is not in the document]
Comment (marcosmol204) [reply package.json:70]: :)

**Comment by mikicho** [comment]
@lirantal 
It seems like last month, we passed the limit. How can we move to the open-source plan?
image

File: src/code-generator/entry-points/non-interactive-cli.ts (hunk 1 above)
Comment (marcosmol204) [reply src/code-generator/entry-points/non-interactive-cli.ts:4]: Nice point, I agree with you. I am pushing a new folder structure.

File: src/code-generator/entry-points/non-interactive-cli.ts [hunk 3]
```diff
@@ -10,17 +11,14 @@ export async function handleNonInteractiveCommand(options: any) {
...
-    📗 Deepen your understanding by reading our article 'Coding with Practica': https://practica.dev/the-basics/coding-with-practica
-    `);
+    spinner.succeed(cliTexts.nonInteractiveCli.onSucceed);
   } catch (error: AppError | any) {
     const errorMessageToUser = error.message
-      ? `❣️ ${error.message}`
-      : `❣️ Embarrassingly our code generator failed. Yeah, almost 100% test coverage did not help here. Could you be nice to us and open an issue?`;
-    console.error(errorMessageToUser);
+      ? `${error.message}`
+      : cliTexts.nonInteractiveCli.onError.default;
```
[the remark this answers is not in the document]
Comment (marcosmol204) [reply src/code-generator/entry-points/non-interactive-cli.ts:20]: :)

**Comment by marcosmol204** [comment]
Pushed updated code. @goldbergyoni

**Comment by lirantal** [comment]
Why is there a private manifest being picked up? this entire project is open sourced no?

**Comment by mikicho** [comment]
@lirantal, maybe when I added the Snyk integration, the repo was private? 
@goldbergyoni Something here is private?

**Comment by lirantal** [comment]
Possibly. I don't actually know as I can't see your account settings. Make sure you login into the Snyk app and look at the settings for projects to see that they're all configured correctly as "from git". 

Helpful resources:
1. https://support.snyk.io/hc/en-us/articles/360000910597-How-can-I-set-a-Snyk-CLI-project-as-open-source-#:~:text=When%20you%20use%20the%20CLI,automatically%20marked%20as%20open%20source.
2. https://support.snyk.io/hc/en-us/articles/360002103557-How-does-Snyk-determine-if-my-project-is-Open-Source-or-not-

**Comment by goldbergyoni** [comment]
@mikicho Nothing is private...

@lirantal Thanks for the help, we owe you a beer which you'll get in NodeTLV 2023

**Comment by lirantal** [comment]
Let me know if you get stuck but should be an easy fix.

**Comment by marcosmol204** [comment]
@goldbergyoni Pushed

**Comment by mikicho** [comment]
@lirantal In an open source like this repo, should I use my **personal** (account) Snyk token or Practica organization (service) account token?

**Comment by lirantal** [comment]
I would say it is preferred to use the Practica one.
By the way, Snyk has a [Snyk for Open Source program](https://snyk.io/open-source-projects/) which grants you a bunch of the pro tier features, perhaps you should look to apply if you aren't yet on this program?

**Comment by mikicho** [comment]
@lirantal I think this is the missing key. maybe because we use my personal key, Snyk limited our tests? or it doesn't matter?

**Review by goldbergyoni — APPROVED** [review]
Looks super great to me

Awesome @marcosmol204

**Pull request merged**